account_invoice_section_sale_order/models/account_move.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class AccountMove(models.Model):
+
+    _inherit = "account.move"
+
+    def _get_ordered_invoice_lines(self):
+        return self.line_ids.sorted(
+            key=self.env["account.move.line"]._get_section_ordering()
+        ).filtered(lambda r: not r.exclude_from_invoice_tab)
+
+
+class AccountMoveLine(models.Model):
+
+    _inherit = "account.move.line"
+
+    def _get_section_group(self):
+        return self.mapped(self._get_section_grouping())
+
+    @api.model
+    def _get_section_grouping(self):
+        invoice_section_grouping = self.env.company.invoice_section_grouping
```

Comment:
seems legit

Reply:
Is there really a case where we could have something different between the invoice we produce and the company from env? I used this because I defined the function as `api.model` to avoid relying on actual move lines records

Reply:
in multi-company mode, if you would have a cron it will be called under main company and invoices may belong to different 
I always prefer to use a company from the obj if there is a special field for it instead of env